iree/compiler/Codegen/Common/FoldAffineMinInDistributedLoops.cpp

```diff
@@ -179,7 +179,13 @@ struct FoldAffineMinInDistributedLoopsPass final
   void runOnOperation() override {
     RewritePatternSet patterns(&getContext());
     populateFoldAffineMinInDistributedLoopsPatterns(patterns);
-    (void)applyPatternsAndFoldGreedily(getOperation(), std::move(patterns));
+    if (failed(applyPatternsAndFoldGreedily(getOperation(),
```

Comment:
This is a new pattern. Maybe this just returning failure when it shouldnt. @antiagainst FYI

Reply:
Interesting. The failure state of `applyPatternsAndFoldGreedily` is really indicating whether the underlying patterns converged or not; so it's not correctness issues. I think here we are having an unfortunate mix between pattern application maybe? Some pattern reverts changes from others? Esp. that the pattern rewriter itself packs in DCE and other folding out of the box..